pkg/ds/daemon_set_test.go

```diff
@@ -295,25 +288,29 @@ func TestSchedule(t *testing.T) {
 	Assert(t).IsNil(err, "Unxpected error trying to mutate daemon set")
 
 	// 11 good nodes 11 bad nodes, and 1 good cherry picked node = 23 nodes
-	numNodes = waitForNodes(t, ds, 23, desiresErrCh, dsChangesErrCh)
-	Assert(t).AreEqual(numNodes, 23, "took too long to schedule")
+	expectedNodes := 23
+	numNodes = waitForNodes(t, ds, expectedNodes, desiresErrCh, dsChangesErrCh)
+	Assert(t).AreEqual(numNodes, expectedNodes, "took too long to schedule")
 
 	//
 	// Deleting the daemon set should unschedule all of its nodes
 	//
+	beforeDeletePods, _, err := scheduledPods(consulStore)
+	Assert(t).IsNil(err, "Unable to list intent/")
 	ds.logger.NoFields().Info("Deleting daemon set...")
 	err = dsStore.Delete(ds.ID())
 	if err != nil {
 		t.Fatalf("Unable to delete daemon set: %v", err)
 	}
-	numNodes = waitForNodes(t, ds, 0, desiresErrCh, dsChangesErrCh)
-	Assert(t).AreEqual(numNodes, 0, "took too long to unschedule")
+	// behavior change: Deleting a daemon set will no longer unschedule its pods (for now)
+	numNodes = waitForNodes(t, ds, 12, desiresErrCh, dsChangesErrCh)
```

Comment:
hey hold on a second. This line waits for the number of nodes to decrease to 12 (spoiler, it never will)